### ISSUE TITLE
Incorrect super c3 linearalization 

### ISSUE THREAD

**Issue by montyly** [crytic/slither#155]
```Solidity
contract A {
    function getValue() public pure returns (uint) {
        return 0;
    }
}

contract B is A {
    function getValue() public pure returns (uint) {
        return 0;
    }
}

contract C is A {
    function getValue() public pure returns (uint) {
        return super.getValue() ;
    }
}

contract E is B, C {
}
```
In this example, slither will fail to understand that in the context of `E`, `return super.getValue() ;` is `        return B.getValue() ;` (and in the context of C only, it is `return A.getValue() ;`

One solution is to create an instance of `getValue` in `E`, but that might break some existing code assumptions.
Another solution is to not translate `super` during the parsing, and to compute it only when needed.

**Comment by montyly** [comment]
Another instance:
```
contract A{
    
    function f1() public{
        f2();
    }
    
    function f2() public{
    }
}

contract B is A{
    function f2() public{
        super.f2();
    }
}
```
`f1()` will incorrecty fix its internal call to `A.f2`, while it will be `B.f2`  in the B context. More I think about it, more I think we should delay the destination resolution of internal functions, and have a context dependent resolver for it.

This can lead to incorrect result and must be fixed prior 0.7.0

**Comment by montyly** [comment]
Fixed in `dev` with #213
